### DISCUSSION TITLE
SEO-Friendly Recipe Site (Schema markup, Structured Data)

### DISCUSSION THREAD

**Discussion by candidexmedia** [GetPublii/Publii#1400] [Ideas]
Hi all!

I'm using this post to continue past discussions on recipe sites with proper structured data: [Plugins for a recipe website](https://forum.getpublii.com/topic/plugins-for-a-recipe-website/) and [Looking For The Best Way To Include Schema Markup](https://forum.getpublii.com/topic/looking-for-the-best-way-to-include-schema-markup/).

Wordpress has quite a few plugins for Recipes which make it easier to add the ingredients + steps, generate structured data, generate recipe cards, aggregated user-submitted star ratings, etc.:

- [Delicious Recipes](https://wordpress.org/plugins/delicious-recipes/)
- [WP Recipe Maker](https://en-ca.wordpress.org/plugins/wp-recipe-maker/)
- etc.

I've gathered current experiments, ideas, and roadblocks below, in the hopes that someone could consolidate everything into a theme or plugin:

## Ingredients Proportions Calculator

[Gerechten, a previous forum poster, mentioned](https://forum.getpublii.com/topic/plugins-for-a-recipe-website/#post-11381) wanting a dynamic way for users to adjust the recipe ingredients for different portions, similar [to this](https://www.leukerecepten.nl/recepten/bladerdeeg-amuse-lepels/):

![](https://i.imgur.com/UPgLfu8.gif)

So I shared (with some GPT help) a similar, rough code which could embedded in a `post.hbs` or [specific post template](https://getpublii.com/dev/how-to-create-custom-templates/) for recipes using the [Post Editor API](https://getpublii.com/dev/post-editor-api/):

![](https://i.imgur.com/XJujAMs.gif)

Source (hit the play ▶️ button to preview): https://editor.p5js.org/candidexmedia/sketches/8CBi0FeEM

## Structured Data

In ordered for recipes to appear as Rich Results in Google, the page needs [structured data](https://developers.google.com/search/docs/appearance/structured-data/recipe) in the markup.

![](https://i.imgur.com/ElGPUXG.png)

The full spec can be found here: https://developers.google.com/search/docs/appearance/structured-data/recipe

I also found the following generators:

- https://webcode.tools/generators/json-ld/recipe
- https://technicalseo.com/tools/schema-markup-generator/
- https://seosmoothie.com/free-seo-tools/schema-markup-generator/recipe

## What's missing: Easy way to add ingredients and steps once

My wish would be to have a post editor option (or popup?) where users can easily enter the ingredients list and steps just once, and then have that information be populated in the appropriate places (`<head>` markup, recipe card, other places on the page, etc.)

I think everything else for the recipe schema can be sourced using Publii [Post tags](https://getpublii.com/dev/post-tags/) and [Featured image tags](https://getpublii.com/dev/featured-image-tags/). I have a [longer writeup on that in a gist](https://gist.github.com/candidexmedia/6304251e0bc2baf7ee07565b4f6fac49).

A potential [advanced schema plugin](https://github.com/GetPublii/Publii/discussions/1389#discussioncomment-8592619) would also help...

## What's missing: Aggregated Star Rating

The Hyvor Talk commenting platform (which has a [Publii plugin](https://marketplace.getpublii.com/plugins/hyvor-talk/) 😉) has a ratings widget which can [generate structured data](https://talk.hyvor.com/docs/ratings#structured-data), [starting at 4$/mo](https://talk.hyvor.com/plans). This is useful for displaying the average rating of a recipe in the Rich Results.

Besides that, though, I've had a lot of trouble finding other star rating platforms for static sites.

@Grosso2020's post on their [php liking system](https://github.com/GetPublii/Publii/discussions/1383) made me think that perhaps that approach was a potential avenue for a star rating system for recipes? I'm looping in @HowToMeetLadies as well in case you have other ideas.

**Comment by Scarfatio** [comment]
A recipe plugin would be amazing for Publii. Creating a recipe site or importing one perhaps from Wp Maker pro would allow you to create a very fast static site

**Comment by Adil512** [comment]
This discussion about the Recipe Schema markup is quite insightful! For those who might need a convenient tool to generate Recipe Schema or any other type of schema markup (like FAQ, Article, Local Business, and more), I recommend checking out https://schemamarkup.app/

**Comment by candidexmedia** [comment]
@GetPublii Seems like my post is attracting spam... 🫠
